client/client.js

```diff
@@ -62,71 +63,97 @@ Client.create = function (message) {
 // Instance methods
 
 // Persist subscriptions and presences when not already persisted in memory
-Client.prototype.storeData = function (message) {
-  var subscriptions = this.subscriptions,
-      presences = this.presences,
-      changed = false,
+Client.prototype.storeData = function (messageIn) {
+  var subCount = Object.keys(this.subscriptions).length,
+      presCount = Object.keys(this.presences).length,
+      logNow = false;
+
+  // Persist the message data, according to type
+  switch(messageIn.op) {
+    case 'unsubscribe':
+    case 'sync':
+    case 'subscribe':
+      logNow = this._storeDataSubscriptions(messageIn, subCount);
+      break;
+
+    case 'set':
+      logNow = this._storeDataPresences(messageIn);
+      break;
+  }
+
+  if (logNow) {
```

Comment:
let's always log for now, whenever there is actual data stored. Once we have this feature working fine, we can remove this logging altogether. 

By the way, another approach to logging samples is simply `Math.random*100 < percentage`